src/tsd/PutDataPointRpc.java

```diff
@@ -292,7 +293,18 @@ public void execute(final TSDB tsdb, final HttpQuery query)
     int queued = 0;
     final List<Deferred<Boolean>> deferreds = synchronous ? 
         new ArrayList<Deferred<Boolean>>(dps.size()) : null;
-    
+
+    if (tsdb.getConfig().enable_header_tag()) {
```

Comment:
For consistency, please use this method for checking the config value: 

 tsdb.getConfig().getString("tsd.http.enable_header_tag").equals("true");